### ISSUE TITLE
False negative with CA1856

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn-analyzers#6824]
### Analyzer

**Diagnostic ID**: [CA1864](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1864) Prefer the 'IDictionary.TryAdd(TKey, TValue)' method

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

### Describe the bug

The analyzer fails to issue a diagnostic when the value argument to Add is a parameter.

### Steps To Reproduce

```C#
    public void Diagnostic(string key)
    {
        int value = 42;
        if (!_data.ContainsKey(key))
        {
            _data.Add(key, value);
        }
    }

    public void NoDiagnostic(string key, int value)
    {
        if (!_data.ContainsKey(key))
        {
            _data.Add(key, value);
        }
    }
```

### Expected behavior

Both calls to `ContainsKey` issue the diagnostic.

### Actual behavior

Only the first call results in a diagnostic.  The second doesn't.
